Connectors/Autocad/Speckle.Connectors.Civil3dShared/HostApp/PropertySetBaker.cs

```diff
@@ -0,0 +1,369 @@
+using Microsoft.Extensions.Logging;
+using Speckle.Connectors.Common.Operations;
+using Speckle.Converters.Civil3dShared;
+using Speckle.Converters.Civil3dShared.Helpers;
+using Speckle.Converters.Common;
+using Speckle.Sdk;
+using Speckle.Sdk.Models;
+using AAEC = Autodesk.Aec;
+using AAECPDB = Autodesk.Aec.PropertyData.DatabaseServices;
+using ADB = Autodesk.AutoCAD.DatabaseServices;
+
+namespace Speckle.Connectors.Civil3dShared.HostApp;
+
+/// <summary>
+/// Helper class to bake property sets to entities on receive.
+/// </summary>
+public class PropertySetBaker
+{
+  private readonly IConverterSettingsStore<Civil3dConversionSettings> _settingsStore;
+  private readonly ILogger<PropertySetBaker> _logger;
+  private readonly PropertyHandler _propertyHandler;
+
+  /// <summary>
+  /// Map of property set definition name to its ObjectId. Populated during ParsePropertySetDefinitions.
+  /// </summary>
+  private readonly Dictionary<string, ADB.ObjectId> _propertySetDefinitionMap = new();
+
+  public PropertySetBaker(
+    IConverterSettingsStore<Civil3dConversionSettings> settingsStore,
+    ILogger<PropertySetBaker> logger
+  )
+  {
+    _settingsStore = settingsStore;
+    _logger = logger;
+    _propertyHandler = new PropertyHandler();
+  }
+
+  /// <summary>
+  /// Parse and bake all property set definitions from the root object.
+  /// Should be called once at the beginning of the receive operation.
+  /// </summary>
+  public void ParsePropertySetDefinitions(Base rootObject)
+  {
+    _propertySetDefinitionMap.Clear();
+
+    if (rootObject[ProxyKeys.PROPERTYSET_DEFINITIONS] is not Dictionary<string, object?> definitions)
+    {
+      return;
+    }
+
+    if (definitions.Count == 0)
+    {
+      return;
+    }
+
+    using var tr = _settingsStore.Current.Document.Database.TransactionManager.StartTransaction();
+
+    foreach (var definition in definitions)
+    {
+      string setName = definition.Key;
+      object? setDefObj = definition.Value;
+
+      if (setDefObj is not Dictionary<string, object?> setDefData)
+      {
+        _logger.LogWarning("Property set definition {SetName} has invalid data format", setName);
+        continue;
+      }
+
+      if (!setDefData.TryGetValue("propertyDefinitions", out var propDefsObj))
+      {
+        _logger.LogWarning("Property set definition {SetName} missing propertyDefinitions", setName);
```

Comment:
missing '$'

Reply:
My two cents on this -> [here](https://github.com/specklesystems/speckle-sharp-connectors/pull/1149#discussion_r2435446247)